js/filter.js

```diff
@@ -48,4 +110,38 @@ export const filterToParams = (options = {}, nameAlias = {}, nameFunc = {}, filt
     return params;
 };
 
+const _buildFilter = (field, arithOp, value, keywordFields) => {
+    if (!value) return [`${field}:${arithOp}:${value}`];
+
+    // verifies if the value is a keyword, if not
+    // returns the filter query with the given value
+    const result = value.match(KEYWORD_REGEX);
+    if (!result || !Object.keys(KEYWORDS).includes(value)) return [`${field}:${arithOp}:${value}`];
+
+    // verifies if the field given allows the usage of
+    // keywords, forbidding invalid usage of keywords
+    // with fields that do not permit it
+    if (keywordFields.length > 0 && !keywordFields.includes(field)) return [];

```

Comment:
I believe you can ditch the `length` verification.

In any case, I'd skip this verification altogether because if we are to do this I believe it needs to be more sophisticated and associate certain keywords with certain fields.

Reply:
Would it make sense to organize keywords by groups and define in pulse the groups each field allows, or simply define which keywords are valid for each field?

Reply:
2nd option 👍

Reply:
Done, the keywords are passed to the filter mixin in pulse's domain mixin